code/src/App.js

```diff
@@ -1,9 +1,79 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import NewThought from 'components/NewThought'
+import HappyThoughtTrain from 'components/HappyThoughtTrain'
 
 export const App = () => {
+  const [happyThoughtTrain, setHappyThoughtTrain] = useState([]);
+  const [newThought, setNewThought] = useState('');
+  const [loading, setLoading] = useState(true);
+
+  const fetchMessage = () => {
+    setLoading(false);
+    fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts')
+      .then((result) => result.json())
+      .then((json) => setHappyThoughtTrain(json))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false));
+  };
+
+  useEffect(() => {
+    fetchMessage();
+  }, [newThought, happyThoughtTrain]);
+
+  const handleNewThoughtChange = (event) => {
+    setNewThought(event.target.value)
+  }
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        message: newThought
+      })
+    }
+
+    fetch('https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts', options)
+      .then((response) => response.json())
+      .then(() => fetchMessage())
+      .catch((error) => console.error(error))
+      .finally(() => setNewThought(''))
+  }
```

Comment:
First I thought the button didn't work but then I realized that it was only because 4 characters cannot be posted. You could add a conditional statement here in case the text is less then 5 characters then show an error message.